mlabs/src/Mlabs/NFT/Governance/Validation.hs

```diff
@@ -63,28 +312,257 @@ govMintPolicy x =
 
 -- | Minting policy for GOV and xGOV tokens.
 mkGovScript :: UniqueToken -> GovDatum -> GovAct -> ScriptContext -> Bool
-mkGovScript _ _ act _ =
+mkGovScript ut _ act ctx =
   case act of
     InitialiseGov ->
-      True
+      False -- FIXME Why should anybody for good reason use it as the redeemer?!
```

Comment:
You need to mint a token for the head when the list is initialised.

Reply:
This token must be in the same eUTXO with the Unique Token to be able to find (on-chain) what the correct minting Policy CurrencySymbol is.

Reply:
The off-chain implementation is currently a bit different
```    -- Mint the Linked List Head and its associated token.
    mintListHead :: NftAppInstance -> GenericContract NftAppInstance
    mintListHead appInstance = do
      let -- Unique Token
          uniqueToken = appInstance'AppAssetClass appInstance
          uniqueTokenValue = assetClassValue uniqueToken 1
          emptyTokenName = TokenName PlutusTx.Prelude.emptyByteString
      let -- Script Head Specific Information
          headDatum :: DatumNft = nftHeadInit appInstance
          headPolicy = mintPolicy appInstance
          proofTokenValue = Value.singleton (scriptCurrencySymbol headPolicy) emptyTokenName 1
          initRedeemer = asRedeemer Initialise
      let -- Gov App Head Specific information
          govHeadDatum :: GovDatum = govHeadInit
          govHeadPolicy = govMintPolicy appInstance
          govScr = govScript uniqueToken
          govProofTokenValue = Value.singleton (scriptCurrencySymbol govHeadPolicy) emptyTokenName 1
          govInitRedeemer = asRedeemer InitialiseGov

          -- NFT App Head
          (lookups, tx) =
            ( mconcat
                [ Constraints.typedValidatorLookups txPolicy
                , Constraints.mintingPolicy headPolicy
                , Constraints.mintingPolicy govHeadPolicy
                ]
            , mconcat
                [ Constraints.mustPayToTheScript headDatum (proofTokenValue <> uniqueTokenValue)
                , Constraints.mustPayToOtherScript (validatorHash govScr) (toDatum govHeadDatum) (govProofTokenValue <> uniqueTokenValue)
                , Constraints.mustMintValueWithRedeemer initRedeemer proofTokenValue
                , Constraints.mustMintValueWithRedeemer govInitRedeemer govProofTokenValue
                ]
            )
      void $ Contract.submitTxConstraintsWith @NftTrade lookups tx
      Contract.logInfo @Hask.String $ printf "Forged Script Head & Governance Head for %s" (Hask.show appInstance)
      return appInstance```
      
Creating the list head and minting the head token are happening at the same transaction. So we are not actually spending the list head when minting the head token. Therefore, the `InitialiseGov` is not being used as the redeemer to the script validator.

Reply:
If you look at 
```
govInitRedeemer = asRedeemer InitialiseGov
```
The redeemer used at initialisation is the `InitialiseGov` redeemer. It is the only redeemer that should allow the minting of a token with `Head` datum.

Reply:
But the `govInitRedeemer` is only used for the minting policy here. Not for the script validator.